snakesist/delb_plugins.py

```diff
@@ -0,0 +1,87 @@
+from types import SimpleNamespace
+from typing import Any, Dict
+
+from delb.plugins import plugin_manager, DocumentExtensionHooks
+from delb.typing import LoaderResult
+
+
+@plugin_manager.register_loader()
+def existdb_loader(source: Any, config: SimpleNamespace) -> LoaderResult:
+    raise NotImplemented
+
+
+@plugin_manager.register_document_extension
+class ExistDBExtension(DocumentExtensionHooks):
+    """
+    This class provides extensions to :class:`delb.Document` in order to interact
+    with a eXist-db instance.
+
+    Documents can be loaded from an eXist-db instance by either specifying an URL with
+    the ``existdb://`` scheme or by specifying a path made up of a collection path and
+    a filename and passing a :class:`snakesist.ExistClient` instance with the
+    ``existdb_client`` keyword. These two initializations would yield the same
+    :class:`delb.Document`:
+
+    .. code-block::
+
+        import delb
+        import snakesist
+
+        # with an URL
+        document = delb.Document(
+            "existdb://example.exist-host.org/exist/collection/document.xml"
+        )
+
+        # with a client
+        client = snakesist.ExistClient(
+            host="example.exist-host.org",
+            prefix="exist",
```

Comment:
that the default though

Reply:
true. I don't think see it as problematic to name it explicitly here though, as it serves the illustrative purpose of the two examples better.

Reply:
stümmt.